compiler/rustc_resolve/src/late/diagnostics.rs

```diff
@@ -985,22 +985,21 @@ impl<'a: 'ast, 'ast> LateResolutionVisitor<'a, '_, 'ast> {
         let ns = source.namespace();
         let is_expected = &|res| source.is_expected(res);
 
-        let path_sep = |err: &mut Diagnostic, expr: &Expr| match expr.kind {
-            ExprKind::Field(_, ident) => {
+        let path_sep = |err: &mut Diagnostic, expr: &Expr| match &expr.kind {
+            ExprKind::Field(base, ident) => {
                 err.span_suggestion(
-                    expr.span,
+                    base.span.source_callsite().between(ident.span),
                     "use the path separator to refer to an item",
-                    format!("{}::{}", path_str, ident),
+                    "::",
                     Applicability::MaybeIncorrect,
                 );
                 true
             }
-            ExprKind::MethodCall(ref segment, ..) => {
-                let span = expr.span.with_hi(segment.ident.span.hi());
+            ExprKind::MethodCall(_, arguments, span) => {
                 err.span_suggestion(
-                    span,
+                    arguments[0].span.source_callsite().between(*span),
```

Comment:
`argument[0]` is the receiver of the method call and it has to exist. However, #100232 changed this (the PR is currently in the Homu queue), so please hold off from r+'ing until it has been merged. I am going to rebase onto master as soon as it lands.

Reply:
Sounds good.